apps/test/unit/javalab/JavalabSettingsTest.js

```diff
@@ -0,0 +1,50 @@
+import {assert} from '../../util/deprecatedChai';
+import React from 'react';
+import {shallow} from 'enzyme';
+import sinon from 'sinon';
+import {JavalabSettings} from '@cdo/apps/javalab/JavalabSettings';
+
+const clickSpy = sinon.spy();
+
+const defaultProps = {
+  style: {},
+  children: [
+    <a onClick={clickSpy}>onclick</a>,
+    <a onClick={clickSpy}>onclick2</a>
+  ]
+};
+
+describe('JavalabSettings', () => {
+  beforeEach(() => clickSpy.resetHistory());
+
+  it('is initially just a button', () => {
+    const wrapper = shallow(<JavalabSettings {...defaultProps} />);
+    assert.strictEqual(wrapper.children().length, 1);
+    assert.strictEqual(wrapper.childAt(0).name(), 'button');
+  });
+
+  it('shows children when clicked', () => {
+    const wrapper = shallow(<JavalabSettings {...defaultProps} />);
+    wrapper.find('button').simulate('click');
```

Comment:
In general, avoid `simulate('click')` on shallow wrapped enzyme elements. It's [discouraged by the writers of enzyme](https://github.com/enzymejs/enzyme/issues/2173). They recommend calling the event handler directly. i.e.
`wrapper.find('button').instance().toggleDropdown()` you may need to call `wrapper.update()` afterward.